src/com/comeet/RoomsDao.java

```
@@ -208,5 +212,107 @@ public static void retrieveMetadata(Room room) {
+     * @return The list of metro areas and their room list names
+     * @throws Exception On an unexpected error.
+     */ 
+    public List<BuildingList> getCriteria(String domain) throws Exception {
+        DataRepository db = new DataRepository();
+        List<BuildingList> result = db.retrieveSearchCriteria(domain);
+        
+        return result;
+    }
+    
+    
+    /**
+     * TODO
+     * @param domain domain of organization to search for
+     * @return The list of metro areas and their room list names
+     * @throws Exception On an unexpected error.
+     */ 
+    private List<EmailAddress> getBuildingRoomlist(String email) throws Exception {
+
+        List<EmailAddress> names = null;
+
+        try {
+            EmailAddressCollection c = service.getRoomLists();
+            for (EmailAddress e : c) {
+                if (e.getAddress().equalsIgnoreCase(email)) {
+                    
+                    names = new ArrayList<EmailAddress>();
+                    Collection<EmailAddress> rooms = service.getRooms(e);
+           
+                    for (EmailAddress r : rooms) {
+                        System.out.println(r.toString());
+                        System.out.println(r.getAddress());
+                        System.out.println(r.getName());
+                        names.add(r);
+                    }
+                }
+            }
+        } catch (ServiceRequestException e) {
+            throw e;
+        } catch (ServiceResponseException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ExchangeServiceException(e);
+        }
+        return names;
+    }
+    
+    /**
+     * Gets all rooms at the organization.
+     * 
+     * @return A list of rooms.
+     */
+    public List<Room> getBuildingRooms(String buildingEmail) throws Exception {
+        List<EmailAddress> rooms = null;
+        try {
+            rooms = getBuildingRoomlist(buildingEmail);
+        } catch (Exception ex) {

```

Don't swallow `Exception`